gengo/src/builder.rs

```diff
@@ -50,22 +48,21 @@ impl<P: AsRef<Path>> Builder<P> {
     }
 
     pub fn build(self) -> Result<Gengo, Box<dyn ErrorTrait>> {
-        let repository = match Repository::discover(self.repository_path) {
+        let repository = match gix::discover(self.repository_path) {
             Ok(r) => r,
-            Err(e) => match e.code() {
-                ErrorCode::NotFound => {
-                    return Err(Box::new(Error::with_source(ErrorKind::NoRepository, e)));
-                }
-                _ => return Err(Box::new(e)),
-            },
+            Err(gix::discover::Error::Discover(err)) => {
```

Comment:
Styling nitpick: import `gix::discover::Error` so this full path doesn't have to be used.